eessi_bot_event_handler.py

```diff
@@ -60,9 +69,137 @@ def handle_issue_comment_event(self, event_info, log_file=None):
+            # next line commented out to write less to log files
+            # self.log(f"searching line '{line}' for bot command")
+            line_stripped = line.strip()
+            if line.isspace() or len(line_stripped) == 0:
+                # next line commented out to write less to log files
+                # self.log(f"line '{line}' is empty")
+                continue
+            bot_command = get_bot_command(line)
+            if bot_command:
+                try:
+                    ebc = EESSIBotCommand(bot_command)
+                except EESSIBotCommandError as bce:
+                    self.log(f"ERROR: parsing bot command '{bot_command}' failed with {bce.args}")
+                    # next two lines commented out to make the bot less noisy,
+                    # could be enabled again if bot commands accept arg --verbose
+                    # or bot instance has a similar setting
+                    # comment_update += f"\n- parsing bot command `{bot_command}` received"
+                    # comment_update += f" from sender `{sender}` failed"
+                    continue
+                commands.append(ebc)
+                self.log(f"found bot command: '{bot_command}'")
+                comment_update += "\n- received bot command "
+                comment_update += f"`{bot_command}`"
+                comment_update += f" from `{sender}` (expanded command format: `{ebc.to_string()}`)"
+            # next two lines are commented out to write less to log files
+            # else:
+                # self.log(f"'{line}' is not considered to contain a bot command")
+                # next three lines commented out to make the bot less noisy,
+                # could be enabled again if bot commands accept arg --verbose
+                # or bot instance has a similar setting
+                # comment_update += f"\n- line <code>{line}</code> is not considered to contain a bot command"
+                # comment_update += "\n- bot commands begin with `bot: `, make sure"
+                # comment_update += "\n  there is no whitespace at the beginning of a line"
+
+        if comment_update == '':
+            # no update to be added, just log and return
+            self.log("update to comment is empty")
+            return
+        else:
+            self.log(f"comment update: '{comment_update}'")
+
+        if not any(map(get_bot_command, comment_update.split('\n'))):
+            # the 'not any()' ensures that the update would not be considered a bot command itself
+            # ... together with checking the sender of a comment update this aims
+            # at preventing the bot to enter an endless loop in commenting on its own
+            # comments
+            update_pr_comment(event_info, comment_update)
+        else:
+            self.log(f"update '{comment_update}' is considered to contain bot command ... not updating PR comment")
+
+        # process commands
+        for cmd in commands:
```

Comment:
The implementation of `handle_issue_comment_event` is getting quite long.

How about introducing two methods there are called in `handle_issue_comment_event`. Something like:

```python
commands = self.extract_bot_commands(...)
self.log("Found %d bot commands in comment ..." % len(commands))
self.process_bot_commands(commands)

Reply:
Hmm, indeed.

However, the data handed over from extracting commands to processing them is not just a list of `commands`. These functions would at least need `event_info`, the comment GitHub instance (or we need to query GitHub a second time), plus possibly other state. If we move them to a separate module, we also need to take of config (now accessible as `self.cfg`), logging is affected ...

I agree it's long. Tend to leave it as is and rather open an issue for refactoring this later.

Reply:
OK, let's open the issue.
I don't think we should move it to a separate module, it can be methods which can access `self.cfg`, but if passing around too much data is required, then maybe it doesn't make much sense.